cpp/net/connection_pool.cc

```
@@ -0,0 +1,62 @@
+#include "net/connection_pool.h"
+
+#include <glog/logging.h>
+
+using std::lock_guard;
+using std::move;
+using std::mutex;
+using std::pair;
+using std::string;
+
+namespace cert_trans {
+namespace internal {
+
+namespace {
+
+const int kMaxConnPerHostPort = 4;
+
+}  // namespace
+
+ConnectionPool::ConnectionPool(libevent::Base* base)
+    : base_(CHECK_NOTNULL(base)) {
+}
+
+
+evhttp_connection_unique_ptr ConnectionPool::Get(const URL& url) {
+  // TODO(pphaneuf): Add support for other protocols.
+  CHECK_EQ(url.protocol(), "http");
+  const pair<string, uint16_t> key(url.host(),
+                                   url.port() != 0 ? url.port() : 80);
+  lock_guard<mutex> lock(lock_);
+
+  auto it(conns_.find(key));
+  if (it == conns_.end()) {
+    return evhttp_connection_unique_ptr(
+        base_->HttpConnectionNew(key.first, key.second));
+  }
+
+  evhttp_connection_unique_ptr retval(move(it->second.back()));
+  it->second.pop_back();
+
+  return retval;
+}
+
+
+void ConnectionPool::Put(evhttp_connection_unique_ptr&& conn) {
+  char* host;
+  uint16_t port;
+  evhttp_connection_get_peer(conn.get(), &host, &port);
```

I think silently ignoring that is the best course of action.